src/main/java/org/myrobotlab/service/interfaces/LocaleProvider.java

```diff
@@ -0,0 +1,36 @@
+package org.myrobotlab.service.interfaces;
+
+import java.util.Map;
+
+import org.myrobotlab.service.data.Locale;
+
+public interface LocaleProvider {
+  
+  /**
+   * set a locale from a code string.
+   * The code string accepts underscore or hyphens
+   * en-US fr-FR or just "en" or "fr" is the country is not needed
+   * @param code
+   * @return
+   */
+  public String setLocale(String code);
```

Comment:
this should return void.

Reply:
ok, although it is satisfying in REST apis to have the value echoed back...
done.

Reply:
Changed, although not only in REST, but additionally this would prevent the value if something was interested in subscribing to it.